app/views/posts/show.html.erb

```diff
@@ -7,7 +7,13 @@
         </div>
     </div>
     <p><%= @post.text %></p>
-    <%= link_to "like", user_post_likes_path(@post.author, @post), class: 'like-button' %>
+    <div class="post-action-buttons">
+        <%= link_to "like", user_post_likes_path(@post.author, @post), class: 'like-button' %>
+        <%= link_to "Delete", user_post_path(@post.author, @post), data: {
+                turbo_method: :delete,
+                turbo_confirm: 'Are you sure you want to delete this post?'
+            }, class: 'like-button' if can? :destroy, Post %>
+    </div>
```

Comment:
- [ ] The delete button for post is displayed regardless of the user's role. Kindly note that only the creator of the post and the admin should be able to see the delete buttons at all. Kindly fix your views.

- [ ] Your post delete functionality does not work at all. When I click it as the admin or post owner, nothing happens - the post is not deleted. Kindly fix that.